### ISSUE TITLE
VoiceOver reads the previous word when navigating by word

### ISSUE THREAD

**Issue by ivanfetch** [microsoft/vscode#87665]
Issue Type: <b>Bug</b>

1. Enable accessibility mode in settings.
2. Enter this text in a new document (including the commas): Hello world, how are you, is your day going well?
3. Press command-leftarrow to move to the beginning of the line - the cursor will be on the letter `h` of the word hello.
4. Press option-rightarrow to move to the next word - the cursor will correctly be on the letter `w` of world, however the previous word, hello, is spoken.
5. Continue using option-rightarrow to navigate by word - when the cursor traverses a comma, two words are spoken, including the correct word - in this example, the words `how` and `is` are correctly spoken when the cursor has moved to them.
6. The cursor continues to speak the previous word when navigating with option-rightarrow, unless words are separated by punctuation.
7. Using option-leftarrow to navigate by word produces inverse results - the correct word is spoken, unless a comma is traversed, in which case two words are spoken.
Expected Behavior: The current word should be spoken as the cursor moves to it, instead of the previous word being spoken.

VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:57:51.166Z)
OS version: Darwin x64 18.7.0

**Comment by isidorn** [comment]
@ivanfetch Thanks for filling this issue. I have tried reproducing and for me the word that I am navigating to is always read out properly. I can not reproduce what you experience: that the previous word is read out. I have tried your exact steps.

The only issue I can reproduce is for right navigation when there are words separated by a comma two words are spoken instead of one. 
Please note that I am using macOS Catalina. I have not changed any default settings of Voice Over. Now I double checked the settings and I see there is under Verbosity > Text > When Moving the Cursor option to control what word to read. However for both values I always get the word that is navigated to.

I can look into fixing the double word being read out, however I would first like to understand better why I can not reproduce the previous word being read out issue that you see.

Some details on how we will fix this: when we detect there is a screen reader we use a custom navigation commands to align with what screen readers expect for Chrome. Since VS Code is built on top of Electron and Chrome and in order to not suprise the screen reader with some customised navigation we "simplify" our navigation. We have tuned this for NVDA on Windows and it seems to work just fine there, however it might need more tunining for VoiceOver on mac. 
Code pointer https://github.com/Microsoft/vscode/blob/780d875d6f806b6bc872b5c501e43c65b4ad5c3d/src/vs/editor/common/controller/cursorWordOperations.ts#L291

fyi @mltony

**Comment by isidorn** [comment]
Let's leave this open and give time to @ivanfetch to reply.

**Comment by ivanfetch** [comment]
I am using macOS Mojave, and do not currently have a Mac running Catalina. Do you happen to have a Mac running Mojave available to test?

@isidorn Were you testing this bug against version 1.41.1 (commit 26076a4de974ead31f97692a0d32f90d735645c0)?

**Comment by isidorn** [comment]
@ivanfetch Yes I tried Mojave and I can reproduce. 
The issue is that we have fine tuned the navigation behavior when we detect a screen reader. This was done mostly for NVDA and my previous comment has more details about why we did it.
This is not what Voice Over expects on Mojave. So I have simply changed this word navigation heuristic to only be done on Windows. This seems to fix this issue completely for me both on Catalina and Mojave.
You can try out vscode insiders tomorrow it should contain the fix for this

Thanks a lot for providing feedback!

**Comment by JacksonKearl** [comment]
With that same text, ` Hello world, how are you, is your day going well?`, I've found stepping by word backwards still causes `you, is` to be read when the cursor reaches `|is`, when it should just be reading `is`.

**Comment by isidorn** [comment]
This works fine with latest macOS BigSur. Thus closing. Adding to this milestone so we get it verified
